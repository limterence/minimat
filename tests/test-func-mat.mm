/* test-func-mat: library of matrix functions */
int main() {
  matrix u;
  u = [1.0, 2.0, -3.0; 4.0, -5.0, 6.0; 7.0, 8.0, -9.0;];
  printmat(mabs(u));    /* absolute values */
  printmat(mexp(u));    /* exponent values */
  printmat(mlog(u));    /* log values */
  printmat(eye(2));     /* identity matrix */
  printmat(diag(u));    /* diagonal of matrix */
  printmat(ones(2,4));  /* matrix of ones */
  printfloat(sum(u));   /* sum of matrix values */
  printfloat(mean(u));  /* mean of matrix values */
  printfloat(norm(u));  /* euclidean norm of matrix values */
  printfloat(min(u));   /* minimum of matrix values */
  printfloat(max(u));   /* maximum of matrix values */
  printfloat(det(u));   /* determinant of matrix */
  printmat(triu(u,0));  /* upper triangular matrix */
  printmat(tril(u,0));  /* lower triangular matrix */
  printmat(adjoint(u)); /* adjoint of matrix */
  printmat(inv(u));     /* inverse of matrix */
  printmat(inv(u) * u); /* check inverse */
}
