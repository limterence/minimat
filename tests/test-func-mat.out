[3 x 3 float]
  1.00   2.00   3.00 
  4.00   5.00   6.00 
  7.00   8.00   9.00 
[3 x 3 float]
  2.72   7.39   0.05 
 54.60   0.01 403.43 
1096.63 2980.96   0.00 
[3 x 3 float]
  0.00   0.69    nan 
  1.39    nan   1.79 
  1.95   2.08    nan 
[2 x 2 float]
  1.00   0.00 
  0.00   1.00 
[1 x 3 float]
  1.00  -5.00  -9.00 
[2 x 4 float]
  1.00   1.00   1.00   1.00 
  1.00   1.00   1.00   1.00 
 11.00   1.22  16.88  -9.00   8.00 -48.00 [3 x 3 float]
  1.00   2.00  -3.00 
  0.00  -5.00   6.00 
  0.00   0.00  -9.00 
[3 x 3 float]
  1.00   0.00   0.00 
  4.00  -5.00   0.00 
  7.00   8.00  -9.00 
[3 x 3 float]
 -3.00  -6.00  -3.00 
 78.00  12.00 -18.00 
 67.00   6.00 -13.00 
[3 x 3 float]
  0.06   0.12   0.06 
 -1.62  -0.25   0.38 
 -1.40  -0.12   0.27 
[3 x 3 float]
  1.00   0.00   0.00 
  0.00   1.00   0.00 
  0.00   0.00   1.00 
